viewing_party/party.py

```diff
@@ -1,23 +1,196 @@
 # ------------- WAVE 1 --------------------
 
+from pickle import NONE
+import copy
+
 def create_movie(title, genre, rating):
-    pass
+    if not (title and genre and rating): # if title or genre or rationg is None return None
+        return None
+    
+    new_movie = {
+        "title": title,
+        "genre": genre,
+        "rating": rating
+    }
+    
+    return new_movie
+
+def add_to_watched(user_data, movie):
+    
+    for watched in user_data:
+        user_data[watched] = [movie]     
+    
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+
+    for watchlist in user_data:
+        user_data[watchlist] = [movie]     
+    
+    return user_data
+
+    
+
+def watch_movie(user_data, title):
+    # will use deep copy of user data to iterate over it
+    # (since we are modifying user_data)
+    user_data_deep_copy = copy.deepcopy(user_data)
+    # if the "title" is in the movie in the watchlist, then add movie to watched and remove from watchlist
+    #if the "title" is not in the movie in the watchlist, then return user_data
+
+    
+    # outer loop goes through list of movies in "watchlist"
+    for movie in user_data_deep_copy["watchlist"]: 
+        if movie["title"] == title:
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)
+
+    return user_data
+
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    if user_data["watched"] == []: # watched list of movies is empty
+            return 0.0

```

Comment:
nice job using a guard clause at the top! one thing we can do to make this a bit more flexible is to check if `user_data["watched"]` is falsy, not just == to []

```suggestion
    if not user_data["watched"]: # watched list of movies is empty
            return 0.0
```